### DISCUSSION TITLE
What are the lifecycle hooks we can use under `when`?

### DISCUSSION THREAD

**Discussion by peterDuot-stationfive** [shipshapecode/shepherd#1792] [Q&A] [answered]
I'm curious to what are the things we can use under `when` I want a hook to know when the tour is completed or when a step is complete.

**Comment by RobbieTheWagner** [comment, marked as answer]
@peterDuot-stationfive I would suggest `tour.on('complete', function() { // do stuff here })` to setup things to do when the tour is complete. For `when` you can see all the events a step triggers here https://github.com/shipshapecode/shepherd/blob/master/src/js/step.js just look for `trigger`.

**Reply by peterDuot-stationfive** [reply]
Cool, thanks @rwwagner90 !
